sectionproperties/pre/sections.py

```diff
@@ -1,259 +1,568 @@
+                active_list.append(points[i_idx])
+                prev_facet = facet
+                continue
+
+            prev_j_idx = prev_facet[
+                1
+            ]  # Look at the last j_idx to test for a break in the chain of edges
+            if (
+                i_idx != prev_j_idx and holes
+            ):  # If there is a break in the chain of edges...
+                if (
+                    active_list == exterior
+                ):  # ...and we are still accumulating on the exterior...
+                    active_list = interiors[
+                        interior_counter
+                    ]  # ... then move to the interior accumulator
+                else:  # ...or if we are already in the interior accumulator...
+                    interior_counter += (
+                        1  # ...then start the next interior accumulator for a new hole.
+                    )
+                    active_list = interiors[interior_counter]
+                active_list.append(points[i_idx])
+            else:
+                active_list.append(
+                    points[i_idx]
+                )  # Only need i_idx b/c shapely auto-closes polygons
+            prev_facet = facet
+
+        exterior_geometry = Polygon(exterior)
+        interior_polys = [Polygon(interior) for interior in interiors]
+        interior_geometry = MultiPolygon(interior_polys)
+        geometry = Geometry(exterior_geometry - interior_geometry)
+        return geometry
+
+    @staticmethod
+    def from_dxf(dxf_filepath: Union[str, pathlib.Path]) -> Geometry:
+        """
+        An interface for the creation of Geometry objects from CAD .dxf files.
+        """
+        return load_dxf(dxf_filepath)
 
-    def create_mesh(self, mesh_sizes):
-        """Creates a quadratic triangular mesh from the Geometry object.
+    def create_facets_and_control_points(self):
+        self.perimeter = None
+        self.perimeter = list(range(len(self.geom.exterior.coords)))
+        self.holes = []
+        self.points = []
+        self.facets = []
+        self.points, self.facets = create_points_and_facets(self.geom)
+        self.control_points = tuple(self.geom.representative_point().coords)
 
-        :param mesh_sizes: A list of maximum element areas corresponding to each region within the
-            cross-section geometry.
-        :type mesh_size: list[float]
+        for hole in self.geom.interiors:
+            hole_polygon = Polygon(hole)
+            self.holes += tuple(hole_polygon.representative_point().coords)
+        return
 
-        :return: Object containing generated mesh data
-        :rtype: :class:`meshpy.triangle.MeshInfo`
+    def compile_geometry(self):  # Alias
+        # pass
+        self.create_facets_and_control_points()
 
-        :raises AssertionError: If the number of mesh sizes does not match the number of regions
+    def create_mesh(self, mesh_size: float):
```

Comment:
this should remain as `mesh_sizes`, see my post in issue #21, as it breaks single geometry vs compound geometry

Reply:
Confirmed. Fixed!